ttml.go

```diff
@@ -422,13 +507,186 @@ func ReadFromTTML(i io.Reader) (o *Subtitles, err error) {
 
 		}
 		s.Lines = append(s.Lines, *l)
-
+		//setting final alignment of entire <p>
+		if finalAlignment != nil {
+			ts.TTMLInStyleAttributes.TextAlign = finalAlignment
+		}
+		if finalExtent != nil {
+			ts.TTMLInStyleAttributes.Extent = finalExtent
+		}
+		if finalOrigin != nil {
+			ts.TTMLInStyleAttributes.Origin = finalOrigin
+		}
+		if finalDisplayAlignment != nil {
+			ts.TTMLInStyleAttributes.DisplayAlign = finalDisplayAlignment
+		}
+		if finalPadding != nil {
+			ts.TTMLInStyleAttributes.Padding = finalPadding
+		}
+		if finalWritingMode != nil {
+			ts.TTMLInStyleAttributes.WritingMode = finalWritingMode
+		}
+		s.InlineStyle = ts.TTMLInStyleAttributes.styleAttributes()
 		// Append subtitle
 		o.Items = append(o.Items, s)
 	}
+
+	//Pushing Inline style from parent tag to every child tag then combining it with child tag's Style to create a new style and then assigning that style to the child tag.
+	customStyleNumber := 1
+	var customStyleMap = make(map[Style]string)
+	for _, item := range o.Items {
+		parentInlineStyle := item.InlineStyle
+		var lines []Line
+		for _, line := range item.Lines {
+			var lineItems []LineItem
+			for _, lineItem := range line.Items {
+				//as parent style may contain only position settings we should not consider that as a new inline style
+				styleAttr := StyleAttributes{
+					TTMLTextAlign:    parentInlineStyle.TTMLTextAlign,
+					TTMLOrigin:       parentInlineStyle.TTMLOrigin,
+					TTMLExtent:       parentInlineStyle.TTMLExtent,
+					TTMLDisplayAlign: parentInlineStyle.TTMLDisplayAlign,
+					TTMLPadding:      parentInlineStyle.TTMLPadding,
+					TTMLWritingMode:  parentInlineStyle.TTMLWritingMode,
+				}
+				styleAttr.propagateTTMLAttributes()
+				if *parentInlineStyle == styleAttr && *lineItem.InlineStyle == (StyleAttributes{}) {
+					lineItems = append(lineItems, lineItem)
+					continue
+				}
+				if *parentInlineStyle != styleAttr && *lineItem.InlineStyle == (StyleAttributes{}) {
+					lineItem.InlineStyle = parentInlineStyle
+				} else if *parentInlineStyle != styleAttr && *lineItem.InlineStyle != (StyleAttributes{}) {
+					//combine both inline style to create a new inline style
+					inlineStyle := combineInlineStyles(parentInlineStyle, *lineItem.InlineStyle)
+					lineItem.InlineStyle = inlineStyle
+				}
+				// if only inline style is present, create a new style from it, as multiple formats (VTT) doesn't support inline + style.
+				if *lineItem.InlineStyle != (StyleAttributes{}) {
+					var inlineStyle *StyleAttributes
+					inlineStyle = lineItem.InlineStyle
+					if lineItem.Style != nil {
+						inlineStyle = combineInlineStyles(lineItem.Style.InlineStyle, *lineItem.InlineStyle)
+					}
+					newStyle := &Style{
+						InlineStyle: inlineStyle,
+					}
+					if lineItem.Style != nil {
+						newStyle.Style = lineItem.Style.Style
+					}
+					newStyle.InlineStyle.propagateTTMLAttributes()
+					ok, id := checkIfCustomStyleExists(customStyleMap, newStyle)
```

Comment:
Could you make this a one liner : `if ok, id := checkIfCustomStyleExists(customStyleMap, newStyle); ok {`